### DISCUSSION TITLE
How to adjust menu background colors?

### DISCUSSION THREAD

**Discussion by aidanmatchette** [hrsh7th/nvim-cmp#1200] [Q&A] [answered]
I can't seem to adjust the background of the cmp menu, it always is a black color. Any help would be appreciated.
<img width="916" alt="Screen Shot 2022-09-27 at 8 21 48 PM" src="https://user-images.githubusercontent.com/92348198/192899323-507dc4a8-0658-4e8b-8c7b-1dde34174b2b.png">

**Comment by fitrh** [comment, marked as answer]
You can use `config.window.completion.winhighlight` and map `Normal` highlight
to something else, by default, it is mapped to `Pmenu`, so you can also override
`Pmenu` highlight

```lua
vim.api.nvim_set_hl(0, "CmpNormal", { bg = "#FF0000" })

require("cmp").setup({
    window = {
        completion = {
            border = "rounded",
            winhighlight = "Normal:CmpNormal",
        }
    }
})
```

**Reply by bassamsdata** [reply]
Hey, thank you for this snippet and it does work, but the problem here it will be hard coded and when changing the theme, I should change this as well.

like I have this black color cmp when I use any transparent background theme.
How can we set it to to take the theme color instead of hard code the color. 

note: I'm using these themes (tokyonight, rose-pine)


https://user-images.githubusercontent.com/105807570/217364915-98ecdb6b-0701-4be2-8e58-9e1429f35211.mov



https://user-images.githubusercontent.com/105807570/217364940-f9d56cd9-55cf-4475-9916-eee2a68cb976.mov


Thank you again.

**Reply by fitrh** [reply]
> but the problem here it will be hard coded and when changing the theme, I should change this as well

Actually, everything is hard coded by colorscheme

> How can we set it to to take the theme color instead of hard code the color.

**_theme color_** = what color?

You can define `CmpNormal` for each colorscheme, the blacks in your video can be adjusted from `window.documentation.winhighlight`

```lua
require("cmp").setup({
    window = {
        completion = {
            border = "rounded",
            winhighlight = "Normal:CmpNormal",
        },
        documentation = {
            winhighlight = "Normal:CmpDocNormal",
        }
    }
})
```

**Reply by bassamsdata** [reply]
thank you for your answer. it got working good when I choose `Normal:Pmenu` `for window.documentation.winhighlight`
tbh, I didn't played alot with the config of cmp regarding the style, however, now I started understanding how it works. 

Thanks again
